packages/devtools_app/lib/src/screens/memory/primitives/class_name.dart

```diff
@@ -0,0 +1,289 @@
+      'isCore and isDartOrFlutter must be exclusive',
+    );
+  }
+
+  HeapClassName.fromClassRef(ClassRef? classRef)
+      : this(
+          library: _library(
+            classRef?.library?.name,
+            classRef?.library?.uri,
+          ),
+          className: classRef?.name ?? '',
+        );
+
+  HeapClassName.fromHeapSnapshotClass(HeapSnapshotClass? theClass)
+      : this(
+          library: _library(
+            theClass?.libraryName,
+            theClass?.libraryUri.toString(),
+          ),
+          className: theClass?.name ?? '',
+        );
+
+  static String _library(String? libName, String? libUrl) {
+    if (libName != null && libName.isNotEmpty) return libName;
+    return libUrl ?? '';
+  }
+
+  final String className;
+  final String library;
+
+  String get fullName => library.isNotEmpty ? '$library/$className' : className;
+
+  bool get isSentinel => className == 'Sentinel' && library.isEmpty;
+
+  /// Whether a class can hold a reference to an object
+  /// without preventing garbage collection.
+  bool get isWeakEntry {
+    // Classes that hold reference to an object without preventing
+    // its collection.
+    const weakHolders = {
+      '_WeakProperty': '${PackagePrefixes.dart}core',
+      '_WeakReferenceImpl': '${PackagePrefixes.dart}core',
+      'FinalizerEntry': '${PackagePrefixes.dart}_internal',
+    };
+
+    if (!weakHolders.containsKey(className)) return false;
+    if (weakHolders[className] == library) return true;
+
+    // If a class lives in unexpected library, this can be because of
+    // (1) name collision or (2) bug in this code.
+    // Throwing exception in debug mode to verify option #2.
+    // TODO(polina-c): create a way for users to add their weak classes
+    // or detect weak references automatically, without hard coding
+    // class names.
+    assert(false, 'Unexpected library for $className: $library.');
+    return false;
+  }
+
+  /// True, if the library does not belong to a package.
+  ///
+  /// I.e. if the library does not have prefix
+  /// `dart:` or `package:`.
+  /// Examples of such classes: Code, Function, Class, Field,
+  /// number_symbols/NumberSymbols, vector_math_64/Matrix4.
+  bool get isPackageless =>
+      library.isEmpty ||
+      (!library.startsWith(PackagePrefixes.dart) &&
+          !library.startsWith(PackagePrefixes.genericDartPackage));
+
+  /// True, if the package has prefix `dart:` or has perfix `package:` and is
+  /// published by Dart or Flutter org.
+  bool get isDartOrFlutter {
+    if (library.startsWith(PackagePrefixes.dart)) return true;
+    if (library.startsWith(PackagePrefixes.flutterPackage)) return true;
+
+    if (!library.startsWith(PackagePrefixes.genericDartPackage)) return false;
+
+    final slashIndex = library.indexOf('/');
+    if (slashIndex == -1) return false;
+    final packageName = library.substring(
+      PackagePrefixes.genericDartPackage.length,
+      slashIndex,
+    );
```

Comment:
You could shorten this up by using `library.split('/').last`

Reply:
Not really: I need a piece between ':' and '\' and the fastest way to do it is `substring`.